src/core/qgsapplication.cpp

```diff
@@ -1434,22 +1440,33 @@ int QgsApplication::systemMemorySizeMb()
   memoryStatus.dwLength = sizeof( MEMORYSTATUSEX );
   if ( GlobalMemoryStatusEx( &memoryStatus ) )
   {
-    return memoryStatus.ullTotalPhys / ( 1024 * 1024 );
+    return memoryStatus.ullTotalPhys / megabyte;
   }
   else
   {
     return -1;
   }
 #elif defined(Q_OS_LINUX)
-  constexpr int megabyte = 1024 * 1024;
   struct sysinfo si;
   sysinfo( &si );
   return si.totalram / megabyte;
-#elif defined(Q_OS_FREEBSD)
-  return -1;
-#elif defined(Q_OS_OPENBSD)
-  return -1;
-#elif defined(Q_OS_NETBSD)
+#elif defined(Q_OS_FREEBSD) || defined(Q_OS_OPENBSD) || defined(Q_OS_NETBSD)
+  unsigned long physmem;
+  size_t size = sizeof( physmem );
+#ifdef HW_PHYSMEM64
+  // NetBSD and OpenBSD have HW_PHYSMEM and HW_PHYSMEM64
+  // Use 64 bit version
+  int mib[2] = {CTL_HW, HW_PHYSMEM64};
+#else
+  // For FreeBSD
+  int mib[2] = {CTL_HW, HW_PHYSMEM};
+#endif
+
```

Comment:
On FreeBSD, sysctl(2) will be called with size=8, but it wants 4.   On a big-endian machine, this will get the wrong answer.    I realize this is messy, but it seems best to ask for oldp/oldsize as int32_t/sizeof(int32_t) or int64_t/sizeof(int64_t) and not try to alias types.   I'd just move the sysctl into the if.